client/lib/interval/index.js

```diff
@@ -38,71 +29,36 @@ export default React.createClass( {
 			EVERY_THIRTY_SECONDS,
 			EVERY_MINUTE,
 		] ).isRequired,
-		pauseWhenHidden: PropTypes.bool,
-		children: PropTypes.element,
-	},
+	};
 
-	getDefaultProps: () => ( {
-		pauseWhenHidden: true,
-	} ),
+	tick = () => {
+		this.props.onTick();
+	};
```

Comment:
as I don't think we'll need to worry about `this` binding here (since we don't rely on it in other parts of Calypso) we can simplify this to just `tick = () => this.props.onTick()` at which point we can probably see that we don't need `this.tick()` at all - instead we can just use `this.props.onTick` wherever we previously had `this.tick`

Reply:
The upside of the `tick` method is that it uses the fresh value of the `onTick` prop. Retrieved at the time of execution, not at the time of scheduling. `setTimeout( this.props.onTick, 5000 )` will use a value that's 5 seconds old and might have changed since, not the current one.

Using a recursive timeout has a similar advantage over `setInterval`: each tick of the loop uses the fresh value of `this.prop.period`. No need to implement `componentWillReceiveProps`.

By the way, how does `setTimeout` prevent accidental infinite loops? I don't understand what kind of accident is necessary to trigger one with `setInterval` 🙂

Reply:
> By the way, how does setTimeout prevent accidental infinite loops?

like a limit in a `while` loop we can more easily encode a kind of "whitelisting" for the loop iteration. it's not automatic but we at least see where the next iteration is explicitly called

two scenarios could be changing props or component dismount